pkg/datarepair/checker/checker.go

```diff
@@ -200,3 +167,113 @@ func contains(a []string, x string) bool {
+	redundancy := pointer.Remote.Redundancy
+	// we repair when the number of healthy files is less than or equal to the repair threshold
+	// except for the case when the repair and success thresholds are the same (a case usually seen during testing)
+	if numHealthy >= redundancy.MinReq && numHealthy <= redundancy.RepairThreshold && redundancy.RepairThreshold != redundancy.SuccessThreshold {
+		if len(missingPieces) == 0 {
+			checker.logger.Warn("Missing pieces is zero in checker, but this should be impossible -- bad redundancy scheme.")
+			return nil
+		}
+
+		// segment needs repair
+		rmtSegInfo.remoteSegmentsNeedingRepair++
+		err = checker.repairQueue.Insert(ctx, &pb.InjuredSegment{
+			Path:       path,
+			LostPieces: missingPieces,
+		})
+		if err != nil {
+			return Error.New("error adding injured segment to queue %s", err)
+		}
+
+		// if same segment previously was irreparable
+		if _, err = checker.irrdb.Get(ctx, []byte(path)); err == nil {
+			err = checker.irrdb.Delete(ctx, []byte(path))
+			if err != nil {
+				zap.L().Error("repair delete failed", zap.Error(err))
+			}
+		}
+	} else if numHealthy < redundancy.MinReq {
+		pathElements := storj.SplitPath(path)
+		// check to make sure there are at least *4* path elements. the first three
+		// are project, segment, and bucket name, but we want to make sure we're talking
+		// about an actual object, and that there's an object name specified
+		if len(pathElements) >= 4 {
+			project, bucketName, segmentpath := pathElements[0], pathElements[2], pathElements[3]
+			lostSegInfo := storj.JoinPaths(project, bucketName, segmentpath)
+			if contains(rmtSegInfo.remoteSegmentInfo, lostSegInfo) == false {
+				rmtSegInfo.remoteSegmentInfo = append(rmtSegInfo.remoteSegmentInfo, lostSegInfo)
+			}
+		}
+
+		rmtSegInfo.remoteSegmentsLost++
+		// make an entry in to the irreparable table
+		segmentInfo := &pb.IrreparableSegment{
+			Path:               []byte(path),
+			SegmentDetail:      pointer,
+			LostPieces:         int32(len(missingPieces)),
+			LastRepairAttempt:  time.Now().Unix(),
+			RepairAttemptCount: int64(1),
+		}
+
+		// add the entry if new or update attempt count if already exists
+		err := checker.irrdb.IncrementRepairAttempts(ctx, segmentInfo)
+		if err != nil {
+			return Error.New("error handling irreparable segment to queue %s", err)
+		}
+	}
+	return nil
+}
+
+// irrepairProcess picks items from irrepairabledb and spawns a repair worker
+func (checker *Checker) irrepairProcess(ctx context.Context) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	limit := 1
+	var offset int64
+	var rmtSegInfo remoteSegmentInfo
+
+	for {
+		seg, err := checker.irrdb.GetLimited(ctx, limit, offset)
+		if err != nil {
+			return err
+		}
+
+		// zero segments returned with nil err
+		if len(seg) == 0 {
+			break
+		}
+
+		err = checker.checkSegmentStatus(ctx, seg[0].GetSegmentDetail(), string(seg[0].GetPath()), &rmtSegInfo)
+		if err != nil {
+			zap.L().Error("repair failed", zap.Error(err))
```

Comment:
don't use global logger, use `checker.logger`

also method `checkSegmentStatus` not repairing so `repair failed` error message make no sense.

Reply:
done